app/controllers/manage_cmr/order_options_controller.rb

```diff
@@ -54,30 +131,95 @@ def create
     end
   end
 
-  def show
+  def deprecate_order_option
     order_option_id = params[:id]
 
+    response = cmr_client.get_order_options(id: order_option_id, provider_id: current_user.provider_id, token: token)
+    if response.error?
+      Rails.logger.error("#{request.uuid} - OrderOptionsController#deprecate_order_option - Retrieving Order Option Error: #{response.clean_inspect}")
+      flash[:error] = response.error_message
+      redirect_back(fallback_location: manage_collections_path)
+      return
+    end
+
+    existing_order_option = response.body.fetch('items', [])[0] unless response.body.fetch('items', []).empty?
+    native_id = existing_order_option.fetch('meta', {}).fetch('native-id', '')
+    updating_order_option = existing_order_option.fetch('umm', {})
+
+    updating_order_option['Deprecated'] = true
+
+    update_response = cmr_client.create_update_order_option(order_option: updating_order_option, provider_id: current_user.provider_id, native_id: native_id, token: token)
+
+    if update_response.success?
+      flash[:success] = 'Order Option was successfully deprecated.'
+    else
+      Rails.logger.error("Deprecate Order Option Error: #{update_response.clean_inspect}")
+      flash[:error] = update_response.error_message
+    end
+    redirect_to order_options_path
+  end
+
+  def legacy_deprecate_order_option
     if echo_provider_token.blank?
       flash[:error] = "Error retrieving echo provider token.  Try logging in with launchpad"
       redirect_back(fallback_location: manage_collections_path)
       return
     end
 
-    response = cmr_client.get_order_option(order_option_id, echo_provider_token)
+    soap_xml_response = echo_client.deprecate_order_options(echo_provider_token, Array.wrap(params[:id]))
+    if soap_xml_response.success?
+      flash[:success] = 'Order Option was successfully deprecated.'
+    else
+      Rails.logger.error("Deprecate Order Option Error: #{soap_xml_response.error_message}")
+      flash[:error] = soap_xml_response.error_message
+    end
+    redirect_to order_options_path
+  end
+
+  def delete_order_option
+    response = cmr_client.remove_order_option(native_id: params[:id], provider_id: current_user.provider_id, token: token)
```

Comment:
I am slightly concerned about us deleting the `order-options`. When we migrated the order-options from legacy services we associated them to collections and services by adding them as part of the data payload, if we delete the order-option from CMR it will still remain on the payload associated between the service and collection. Querying for an order-option with the concept-id in that payload will of course return no results.

Reply:
I don't think that is necessarily part of this ticket's scope though